src/Microsoft.Dnx.Compilation.CSharp/RoslynCompiler.cs

```diff
@@ -252,58 +252,44 @@ private CompilationModules GetCompileModules(CompilationTarget target)
             });
         }
 
-        private static CSharpCompilation ApplyVersionInfo(CSharpCompilation compilation, CompilationProjectContext project,
+        private static CSharpCompilation ApplyProjectInfo(CSharpCompilation compilation, CompilationProjectContext project,
             CSharpParseOptions parseOptions)
         {
-            const string assemblyFileVersionName = "System.Reflection.AssemblyFileVersionAttribute";
-            const string assemblyVersionName = "System.Reflection.AssemblyVersionAttribute";
-            const string assemblyInformationalVersion = "System.Reflection.AssemblyInformationalVersionAttribute";
+            var attributeNameToProjectValue = new Dictionary<string, string>(StringComparer.Ordinal)
```

Comment:
I don't see any checks for null or invalid values. What happens if the language is null or empty? same for copyright and description.